systemtest/src/test/java/io/strimzi/systemtest/specific/SpecificST.java

```diff
@@ -100,20 +110,151 @@ void testRackAware() {
+                .editSpec()
+                    .withNewRack()
+                        .withTopologyKey(wrongRackKey)
+                    .endRack()
+                .addToConfig("key.converter.schemas.enable", false)
+                .addToConfig("value.converter.schemas.enable", false)
+                .addToConfig("key.converter", "org.apache.kafka.connect.storage.StringConverter")
+                .addToConfig("value.converter", "org.apache.kafka.connect.storage.StringConverter")
+                .endSpec().build());
+
+        PodUtils.waitForPendingPod(CLUSTER_NAME + "-connect");
+        List<String> connectWrongPods = kubeClient().listPodNames(Labels.STRIMZI_KIND_LABEL, KafkaConnect.RESOURCE_KIND);
+        String connectWrongPodName = connectWrongPods.get(0);
+        LOGGER.info("Waiting for ClusterOperator to get timeout operation of incorrectly set up KafkaConnect");
+        TestUtils.waitFor("Wait for KafkaConnect 'Unschedulable' condition due to wrong rackKey used.",
+                Constants.GLOBAL_POLL_INTERVAL, CO_OPERATION_TIMEOUT_SHORT * 2, () -> {
+                List<Condition> conditions = KafkaConnectResource.kafkaConnectClient().inNamespace(NAMESPACE).withName(CLUSTER_NAME).get().getStatus().getConditions();
+                for (Condition condition : conditions) {
+                    if (condition.getReason().matches("TimeoutException") && condition.getType().matches("NotReady")) {
+                        return true;
+                    }
+                }
+                return false;
+            });
+
+        KafkaConnect kc = KafkaConnectResource.kafkaConnectClient().inNamespace(NAMESPACE).withName(CLUSTER_NAME).get();
+        Condition kcCondition = kc.getStatus().getConditions().get(0);
+        assertThat("NotReady", is(kcCondition.getType()));
+        assertThat("TimeoutException", is(kcCondition.getReason()));
+
+        PodStatus kcWrongStatus = kubeClient().getPod(connectWrongPodName).getStatus();
+        assertThat("Unschedulable", is(kcWrongStatus.getConditions().get(0).getReason()));
+        assertThat("PodScheduled", is(kcWrongStatus.getConditions().get(0).getType()));
+
+        KafkaConnectResource.replaceKafkaConnectResource(CLUSTER_NAME, kafkaConnect -> {
+            kafkaConnect.getSpec().setRack(new Rack(rackKey));
+        });
+        KafkaConnectUtils.waitForConnectReady(CLUSTER_NAME);
+        LOGGER.info("KafkaConnect is ready with changed rack key.");
+        LOGGER.info("Verify KafkaConnect rack key update");
+        kc = KafkaConnectResource.kafkaConnectClient().inNamespace(NAMESPACE).withName(CLUSTER_NAME).get();
+        assertThat(kc.getSpec().getRack().getTopologyKey(), is(rackKey));
+
+        if (Environment.DEFAULT_TO_DENY_NETWORK_POLICIES.equals(Boolean.TRUE.toString())) {
+            KubernetesResource.allowNetworkPolicySettingsForResource(kc, KafkaConnectResources.deploymentName(kc.getMetadata().getName()));
+        }
```

Comment:
isn't it by default applied for each KC which we create during tests?

Reply:
No, as I am not using specifically `KafkaConnectResource.deployKafkaConnect()` method. @im-konge also mentioned me this today.

Reply:
I don't say you have to change it in this PR, but maybe we should think about applying the NP in `kafkaConnectWithoutWait` as we do in `deployKafkaConnect`

Reply:
Yes, but I thought that you will add it to the `KafkaConnectResource.kafkaConnectWithoutWait()` method. I don't think that having it directly in a test is a good idea. But ... there comes another problem -> you **have** to create KafkaClients pod before creating the KafkaConnect, so you'll have to add it to all tests where the Connect is deployed via the `withoutWait` method.

Reply:
Ye, maybe we should think more about it and solve it a little bit differently to avoid the situation with clients

Reply:
And you are using this `if` more than once ... so ... what about to create some different method, which will create the NP and then deploy the Connect without wait? To not add unnecessary Clients pods to other testcases?